test/kafo/data_types/integer_test.rb

```diff
@@ -17,7 +17,7 @@ module DataTypes
       describe "#typecast" do
         it { _(Integer.new.typecast(1)).must_equal 1 }
         it { _(Integer.new.typecast('1')).must_equal 1 }
-        it { _(Integer.new.typecast('1foo')).must_equal 1 }
+        it { _(Integer.new.typecast('1foo')).must_equal '1foo' }
```

Comment:
Another difference is `'1.5'`. Previously that became `1` but is now `'1.5'` since it's not a pure integer. I'd also add a testcase for `' 1 '`. It appears to strip so that is good, but I like covering that in tests.

Reply:
Are you saying we should be converting floats to integers? or arguing that unless its truly an integer (or only an integer inside a string) is when we should cast it?

Reply:
I don't think we should convert it, but we should at least have a test case that explicitly describes the behavior.